### PR TITLE
Feature: Minimal manifests 

### PR DESCRIPTION
**Pull request by meggm**
# Description
Need for a simple and minimal manifest file using existing Operator CRD.
Changes include adding the below default values in the code:

- reverse proxy module for PowerMax driver
- disable sdc-monitor and health-monitor by default 
- added the default check for kubelet config directory 
- fixed issues with initContainer and MDM key not found issue in PowerFlex driver

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1449 |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested by installing the drivers against the kubernetes cluster
- [x] Ran tests

### CODE REVIEW

**Review by tdawe — APPROVED** [review]
LGTM

**Review by mgandharva — DISMISSED** [review]

**Review by harishp8889 — APPROVED** [review]

**Review by bandak2 — DISMISSED** [review]

**Review by HarishH-DELL — APPROVED** [review]

File: operatorconfig/driverconfig/powermax/v2.12.0/controller.yaml
Comment (nitesh3108) [inline operatorconfig/driverconfig/powermax/v2.12.0/controller.yaml:1]: Is this intentional creating 2.12.0 structure only for powermax and not for other drivers.

Comment (harishp8889) [reply operatorconfig/driverconfig/powermax/v2.12.0/controller.yaml:1]: @nitesh3108 for powermax alone there are some driver changes and we need to use nightly image for now until the new version is released and hence I have created the latest version for powermax alone and used nightly image in that. Also there are additional params added in the yamls which are only supported in the latest driver

File: samples/storage_csm_powerstore_v2110.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ spec:
     forceUpdate: false
     forceRemoveDriver: true
     common:
-      image: "dellemc/csi-powerstore:v2.11.0"
+      image: "dellemc/csi-powerstore:nightly"
```
Comment (harishp8889) [inline samples/storage_csm_powerstore_v2110.yaml]: @gilltaran why this is nightly image? Please change it to `v2.11.0`

**Review by harishp8889 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**